samples/nrf_desktop/src/events/config_event.h

```diff
@@ -20,33 +20,71 @@
 extern "C" {
 #endif
 
-#define DATA_SIZE_USER_CONFIG 4
 
-#if (REPORT_SIZE_USER_CONFIG <= DATA_SIZE_USER_CONFIG)
-#error "Invalid size of report"
-#endif
+#define MAX_CONFIG_DYNDATA_SIZE 32
+
+/* Config event ID macros */
+#define GROUP_FIELD_POS		6
+#define GROUP_FIELD_SIZE	2
+#define GROUP_FIELD_MASK	BIT_MASK(GROUP_FIELD_SIZE)
+#define GROUP_FIELD_SET(group)	((group & GROUP_FIELD_MASK) << GROUP_FIELD_POS)
+#define GROUP_FIELD_GET(event_id) ((event_id >> GROUP_FIELD_POS) & GROUP_FIELD_MASK)
+
+#define TYPE_FIELD_POS		0
+#define TYPE_FIELD_SIZE		6
+#define TYPE_FIELD_MASK		BIT_MASK(TYPE_FIELD_SIZE)
+#define TYPE_FIELD_SET(type)	((type & TYPE_FIELD_MASK) << TYPE_FIELD_POS)
+#define TYPE_FIELD_GET(event_id) ((event_id >> TYPE_FIELD_POS) & TYPE_FIELD_MASK)
+
+#define CONFIG_EVENT_ID(group, type) ((u8_t)(GROUP_FIELD(group) | \
+					     TYPE_FIELD(type)))
+
+#define EVENT_GROUP_SETUP	0x1
+#define EVENT_GROUP_DFU		0x2
+
+
+/* Config event, setup group macros */
+
+#define MOD_FIELD_POS		3
+#define MOD_FIELD_SIZE		3
+#define MOD_FIELD_MASK		BIT_MASK(MOD_FIELD_SIZE)
+#define MOD_FIELD_SET(module)	((module & MOD_FIELD_MASK) << MOD_FIELD_POS)
+#define MOD_FIELD_GET(event_id) ((event_id >> MOD_FIELD_POS) & MOD_FIELD_MASK)
+
+#define OPT_FIELD_POS		0
+#define OPT_FIELD_SIZE		3
+#define OPT_FIELD_MASK		BIT_MASK(OPT_FIELD_SIZE)
+#define OPT_FIELD_SET(option)	((option & OPT_FIELD_MASK) << OPT_FIELD_POS)
+#define OPT_FIELD_GET(event_id) ((event_id >> OPT_FIELD_POS) & OPT_FIELD_MASK)
+
+#define SETUP_EVENT_ID(module, option) CONFIG_EVENT_ID(EVENT_GROUP_SETUP, \
+						       MOD_FIELD(module) | OPT_FIELD(option))
+
+#define SETUP_MODULE_SENSOR	0x1
+#define SETUP_MODULE_LED	0x2
+
+
+/* Config event, setup group, sensor module macros */
+#define SENSOR_OPT_CPI			0x0
+#define SENSOR_OPT_DOWNSHIFT_RUN	0x1
+#define SENSOR_OPT_DOWNSHIFT_REST1	0x2
+#define SENSOR_OPT_DOWNSHIFT_REST2	0x3
+#define SENSOR_OPT_COUNT 4
```

Comment:
align with others ?

Reply:
this was left in purpose as it is count not the id